website/static/js/filepage/index.js

```diff
@@ -17,6 +18,53 @@ var Panel = utils.Panel;
 
 var EDITORS = {'text': FileEditor};
 
+var SharePopover =  {
+    view: function(ctrl, params) {
+        var copy_button_height = '34px';
+        var popover_width = '450px';
+        var link = params.link.substring(0, params.link.indexOf('download') + 8) + '%26mode=render';
+        var url = link.substring(0, link.indexOf('render'));
+        return m('button.btn.btn-sm.btn-primary.file-share', {onclick: function () {
+                m.render(document.getElementById('popOver'), [
+                    m('ul.nav.nav-tabs.nav-justified', [
+                        m('li.active', m('a[href="#share"][data-toggle="tab"]', 'Share')),
+                        m('li', m('a[href="#embed"][data-toggle="tab"]', 'Embed'))
+                    ]), m('br'),
+                    m('.tab-content', [
+                        m('.tab-pane.fade.in.active#share', m('.input-group', [
+                            m('span.input-group-btn', m('button#copyBtn.btn.btn-default.btn-md[type="button"][style="height:' + copy_button_height + '"][data-clipboard-text="' + link + '"]', m('.fa.fa-copy'))),
+                            m('input.form-control[readonly][type="text"][value="'+ link +'"]')
+                        ])),
+                        m('.tab-pane.fade#embed', [
+                            m('p', 'Dynamically Render iFrame with JavaScript'),
+                            m('textarea.form-control[readonly][type="text"][value="' +
+                                '<script>window.jQuery || document.write(\'<script src="//code.jquery.com/jquery-1.11.2.min.js">\\x3C/script>\') </script>'+
+                                '<link href="' + url + 'static/css/mfr.css" media="all" rel="stylesheet">' +
+                                '<div id="mfrIframe" class="mfr mfr-file"></div>' +
+                                '<script src="' + url + 'static/js/mfr.js">' +
+                                '</script> <script>' +
+                                    'var mfrRender = new mfr.Render("mfrIframe", "' + link + '");' +
+                                '</script>' + '"]'
+                            ), m('br'),
+                            m('p', 'Direct iFrame with Fixed Height and Width'),
+                            m('textarea.form-control[readonly][value="' +
+                                '<iframe src="' + link + '" width="100%" scrolling="yes" height="' + params.height + '" marginheight="0" frameborder="0" allowfullscreen webkitallowfullscreen>"]'
+                            )
+                        ])
+                    ])
+                ]);
+                makeClient($('#copyBtn'));
```

Comment:
Rather than calling makeClient here, you can separate the copy button into its own component

```js
m.component(CopyButton)
```